### PR TITLE
feat: allow to upload price of barcode-less products

### LINKED ISSUE #47: Add price for a product without EAN code
### Story

Currently only prices with a EAN code are allowed

### PR DESCRIPTION
**Pull request by raphael0202**
Add support for barcode-less product by adding a new `category_tag` field in the `prices` table.

Also:
- fix a recursion error in PriceCreate model, due to date namespace clash
- document all fields of PriceCreate model

Fixes #47.

### CODE REVIEW

File: app/schemas.py [hunk 1]
```diff
@@ -84,12 +129,24 @@ def serialize_currency(self, currency: Currency, _info):
             return currency.code
         return currency
 
+    @model_validator(mode="after")
+    def product_code_and_category_tag_are_exclusive(self):
```
Comment (raphodn) [inline app/schemas.py:133]: :+1:

File: app/schemas.py [hunk 2]
```diff
@@ -56,20 +57,64 @@ class LocationBase(LocationCreate):
...
-    date: date
-    proof_id: int | None = None
+    product_code: str | None = Field(
+        default=None,
+        min_length=1,
+        pattern="^[0-9]+$",
+        description="barcode (EAN) of the product, as a string.",
+        examples=["16584958", "1234567890123"],
+    )
+    category_tag: str | None = Field(
```
Comment (raphodn) [inline app/schemas.py:73]: in the long terme could this be an enum from the openfoodfacts-python library ?

File: app/schemas.py [hunk 3]
```diff
@@ -56,20 +57,64 @@ class LocationBase(LocationCreate):
...
+        default=None,
+        min_length=3,
+        pattern=r"^[a-z]{2,}:[a-zA-Z\-]+$",
```
Comment (raphodn) [inline app/schemas.py:76]: we should add tests (not necessarily in this PR if you want to merge quickly)

**Review by raphodn — APPROVED** [review]

File: app/schemas.py (hunk 2 above)
Comment (raphael0202) [reply app/schemas.py:73]: Not really, as the category taxonomy is frequently updated.

File: app/schemas.py (hunk 3 above)
Comment (raphael0202) [reply app/schemas.py:76]: I tested it locally and it works fine, but we definitely need some :)

Comment (raphodn) [reply app/schemas.py:76]: added here #60

**Pull request merged**